### ISSUE TITLE
Metricsql: default should apply of result of the left exp not to entire respose

### ISSUE THREAD

**Issue by tenmozes** [VictoriaMetrics/VictoriaMetrics#5071]
### Describe the bug

Query:

```metricsql
sum by (job) (rate(node_cpu_seconds_total)) > 20 default 5
```

has 2 time series one in higher 20 another one lower 20, when default is applied it affect series which is filtered out by >20.

### To Reproduce

https://play.victoriametrics.com/select/accounting/1/6a716b0f-38bc-4856-90ce-448fd713e3fe/prometheus/graph/#/?g0.range_input=30m&g0.end_input=2023-09-25T20%3A54%3A18&g0.tab=0&g0.relative_time=last_30_minutes&g0.expr=sum+by+%28job%29+%28rate%28node_cpu_seconds_total%29%29+%3E+20+default+5&g1.expr=sum+by+%28job%29+%28rate%28node_cpu_seconds_total%29%29+%3E+20&g1.range_input=30m&g1.end_input=2023-09-25T20%3A54%3A18&g1.tab=0&g1.relative_time=last_30_minutes

### Version

1.93.5-ent (playground version)

**Comment by valyala** [comment]
This is expected behavior - `q default N` replaces all the gaps in `q` series with `N`, even if the `q` series is completely empty because it has been filtered out by some comparison operator. For example, the following query returns all the `temperature` series and replaces all the values bigger than 30 with 40 in all the returned series, even if these series had zero values smaller than 30 initially on the selected time range:

```metricsql
(temperature <= 30) default 40
```

The commit bdb743c88d737d18a5c7e6bec325deee0f35cd8a adds [drop_empty_series(https://docs.victoriametrics.com/MetricsQL.html#drop_empty_series)] function, which can be used for dropping empty series before applying the remaining calculations. For example, the following query doesn't return series with all the initial values bigger than 30 on the selected time range:

```metricsql
drop_empty_series(temperature <= 30) default 40
```

This commit will be included in the next release. In the mean time it is possible to check how this function works by running VictoriaMetrics built from this commit according to [these instructions](https://docs.victoriametrics.com/#how-to-build-from-sources).

**Comment by valyala** [comment]
The [drop_empty_series](https://docs.victoriametrics.com/MetricsQL.html#drop_empty_series) function is available starting from the release [v1.95.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as done.
